### PR TITLE
backend/enhance/revenue_apis_sz

### PR DESCRIPTION
**Pull request by 0utkarsh**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

### CODE REVIEW

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/Clickhouse/DriverFee.hs [hunk 1]
```diff
@@ -132,12 +136,13 @@ findAllByDate merchantId statuses mbFrom mbTo dayBasis mbCollBy = do
                       CH.+. CH.unsafeCoerceNum @(Maybe Int) @(Maybe Centesimal) driverFee.govtCharges
```
Comment (roman-bodavskiy) [inline Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/Clickhouse/DriverFee.hs:136]: don't we need to add `specialZoneAmount` to `totalAmount`?

File: flake.nix [hunk 2]
```diff
@@ -6,7 +6,7 @@
 
     # Backend inputs
     shared-kernel = {
-      url = "github:nammayatri/shared-kernel";
+      url = "path:/Users/utkarsh.pandey1/Desktop/shared-kernel"; # "github:nammayatri/shared-kernel";
```
Comment (roman-bodavskiy) [inline flake.nix:9]: please remove before merge

**Review by roman-bodavskiy — APPROVED** [review]

File: Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/Clickhouse/DriverFee.hs (hunk 1 above)
Comment (0utkarsh) [reply Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/Clickhouse/DriverFee.hs:136]: `totalAmount = openMarketAmount + specialZoneAmount`

File: flake.lock [hunk 3]
```diff
@@ -735,11 +735,14 @@
         "tinylog": "tinylog"
       },
       "locked": {
+        "lastModified": 1709647267,
```
Comment (hkmangla) [inline flake.lock:738]: Why these fields are repeated?

**Review by hkmangla — APPROVED** [review]

**Pull request merged**